ctapipe/image/extractor.py

```diff
@@ -386,15 +409,239 @@ class BaselineSubtractedNeighborPeakWindowSum(NeighborPeakWindowSum):
+        -------
+        charge : array_like
+            Integrated charge per pixel.
+            Shape: (n_pix, n_channels)
+        pulse_time : array_like
+            Samples in which the waveform peak has been recognized.
+            Shape: (n_pix)
+
+        """
+
+        # STEP 1
+
+        # Starting from DL0, the channel is already selected (if more than one)
+        # event.dl0.tel[tel_id].waveform object has shape (N_pixels, N_samples)
+
+        # For each pixel, we slide a 3-samples window through the
+        # waveform without touching the extremes (so later we can increase it
+        # to 5), summing each time the ADC counts contained within it.
+
+        # 'width' could be configurable in a generalized version
+        # Right now this image extractor is optimized for LSTCam and NectarCam
+        width = 3
+        sums = np.apply_along_axis(slide_window, 1, waveforms[:, 1:-1], width)
+        # Note that the input waveforms are clipped at the extremes because
+        # we want to extend this 3-samples window to 5 samples
+        # 'sums' has now the shape of (N_pixels, N_samples-4)
+
+        # For each pixel, in each of the (N_samples - 4) positions, we check
+        # where the window encountered the maximum number of ADC counts
+        startWindows = np.apply_along_axis(np.argmax, 1, sums)
+        # Now startWindows has the shape of (N_pixels).
+        # Note that the index values stored in startWindows come from 'sums'
+        # of which the first index (0) corresponds of index 1 of each waveform
+        # since we clipped them before.
+
+        # Since we have to add 1 sample on each side, window_shift will always
+        # be (-)1, while window_width will always be window1_width + 1
+        # so we the final 5-samples window will be 1+3+1
+        window_widths = np.full_like(startWindows, width + 1)
+        window_shifts = np.full_like(startWindows, 1)
+
+        # the 'peak_index' argument of 'extract_around_peak' has a different
+        # meaning here: it's the start of the 3-samples window.
+        # Since since the "sums" arrays started from index 1 of each waveform,
+        # then each peak index has to be increased by one
+        charge_1stpass, pulse_time_1stpass = extract_around_peak(
+            waveforms, startWindows + 1, window_widths, window_shifts
+        )
+
+        # STEP 2
+
+        # Set thresholds for core-pixels depending on telescope type.
+        # Boundary thresholds will be half of core thresholds.
+        # WARNING: these values should be read from a configuration file
+        # à-la-protopipe and they will depend in principle also on camera type.
+        subarray = self.subarray
+        if subarray.tel[telid].type == "LST":
+            core_th = 6  # (not yet optimized)
+        if subarray.tel[telid].type == "MST":
+            core_th = 8  # (not yet optimized)
+        if subarray.tel[telid].type == "SST":
+            core_th = 4  # (not yet optimized)
+
```

Comment:
This should just be a `TelescopeParameter` in the class, so that it can be configured from the config file. 
 
at the top of the class, and you can even set these values as the default:
```py3
core_threshold = FloatTelescopeParameter(
    default_value=[('type','*',6.0), ('type','LST*',6.0), ('type','MST*',8.0), ('type','SST*',4.0)],
    help="Picture threshold for internal tail-cuts pass"
)
```
Then here you can just use:
 ```py3
core_th = self.core_threshold.tel[telid]
```

and it will use the correct threshold

Reply:
Thanks, done!